lib/miq_ldap.rb

```diff
@@ -445,7 +445,9 @@ def self.default_ldap_port(scheme = "ldap")
   end
 
   def self.using_ldap?
-    ::Settings.authentication.mode.include?('ldap')
+    ::Settings.authentication.mode.include?('ldap').tap do |should_warn|
```

Comment:
odd place to do log the warning, isn't using_ldap? used in many places so we'd end up with tons of those deprecation warnings ?

Reply:
I traced all the methods in miq_ldap.rb and `using_ldap?` was only invoked once pre login. So I thought that would be the best place. I had also considered issuing the warning only when setting the mode but that would mean the deprecation would not be coming from the MiqLdap code that will be deleted. I had discussed with @gtanzillo  and we agreed, although not totally necessary,  it might be best to issue the warning in the code that will eventually be deleted.

Reply:
@abellotti I just checked and it’s also only issued once per API authentication.

Reply:
@abellotti As we discussed in chat. I have tested with the API both with and without using a token and the warning is only issued once per authentication. The warning is not reused when the token is used with the API. The warning is only issued once, when acquiring the token.